test/CMakeLists.txt

```diff
@@ -1,32 +1,48 @@
+cmake_minimum_required(VERSION 3.14...3.25)
```

Comment:
Why limit the maximum?

Reply:
Because of CMake policies which came out in the future. CMake knows which policies to use.